src/ftp-deploy.js

```diff
@@ -53,8 +53,9 @@ const FtpDeployer = function () {
                     .put(tmp, upath.join(config.remoteRoot, relDir, fname))
                     .then(() => {
                         this.eventObject.transferredFileCount++;
+                        this.eventObject.filename = fname;
                         this.emit('uploaded', this.eventObject);
-                        return Promise.resolve("uploaded " + tmpFileName);
+                        return Promise.resolve("uploaded " + this.eventObject.filename);
```

Comment:
this would return the remote path+name rather than the local one I think? I'm not sure which is better

Reply:
I think in my case `tmpFileName` was not returning anything. It may have something to do with using `map()` instead of `mapSeries()`.

Reply:
there is clearly something different around Promises betwen us, as you were experiencing issues with mapSeries that I did not see. Indeed mapSeries is vital for me because your code would create an arbitrary number of concurrent ftp connections, and I know that the servers I use do not permit that